packages/api/src/command/medical/patient/update-patient.ts

```diff
@@ -22,42 +24,130 @@ export type PatientUpdateCmd = BaseUpdateCmdWithCustomer &
-    patientDiscovery: { requestId, startedAt: new Date() },
-  };
+  await getFacilityOrFail({ cxId, id: facilityId });
 
-  const patient = await updatePatientWithoutHIEs(patientUpdateWithPD, emit);
+  const patient = await updatePatientWithoutHIEs(patientUpdate, emit);
 
   const fhirPatient = toFHIR(patient);
   await upsertPatientToFHIRServer(patientUpdate.cxId, fhirPatient);
 
-  await cqCommands.patient.discover(patient, facility.id, requestId, forceCarequality);
-
-  await cwCommands.patient.update(
+  const updatedPatient = await scheduleOrRunPatientDiscovery({
     patient,
     facilityId,
-    getCqOrgIdsToDenyOnCw,
-    requestId,
-    forceCommonwell
-  );
+    rerunPdOnNewDemographics,
+    augmentDemographics,
+    forceCommonwell,
+    forceCarequality,
+  });
+
+  return updatedPatient;
+}
+
+// Needs to live somewhere else
+export async function scheduleOrRunPatientDiscovery({
+  patient,
+  facilityId,
+  rerunPdOnNewDemographics,
+  augmentDemographics,
+  forceCommonwell,
+  forceCarequality,
+  isRerunFromNewDemographics = false,
+  overrideSchedule = false,
+}: {
+  patient: Patient;
+  facilityId: string;
+  rerunPdOnNewDemographics: boolean;
+  augmentDemographics: boolean;
+  // START TODO #1572 - remove
+  forceCommonwell?: boolean;
+  forceCarequality?: boolean;
+  // END TODO #1572 - remove
+  overrideSchedule?: boolean;
+  isRerunFromNewDemographics?: boolean;
+}): Promise<Patient> {
+  let updatedPatient = patient;
+  const cqData = cqCommands.patient.getCQData(patient.data.externalData);
+
+  const discoveryStatusCq = cqData?.discoveryStatus;
+  const scheduledPdRequestCq = cqData?.scheduledPdRequest;
+
+  if (
+    discoveryStatusCq === "processing" &&
+    (!scheduledPdRequestCq || (scheduledPdRequestCq && overrideSchedule))
+  ) {
+    updatedPatient = await schedulePatientDiscovery({
+      requestId: uuidv7(),
+      patient,
+      source: MedicalDataSource.CAREQUALITY,
+      facilityId,
+      rerunPdOnNewDemographics,
+      augmentDemographics,
+      isRerunFromNewDemographics,
+    });
+  } else if (discoveryStatusCq === "processing" && scheduledPdRequestCq) {
+    // Do nothing
+  } else {
+    await cqCommands.patient.discover({
+      patient,
+      facilityId,
+      forceEnabled: forceCarequality,
+      rerunPdOnNewDemographics,
+      augmentDemographics,
+    });
+  }
+
+  const cwData = cwCommands.patient.getCWData(patient.data.externalData);
+
+  const statusCw = cwData?.status;
+  const scheduledPdRequestCw = cwData?.scheduledPdRequest;
+
+  if (
+    statusCw === "processing" &&
+    (!scheduledPdRequestCw || (scheduledPdRequestCw && overrideSchedule))
+  ) {
+    updatedPatient = await schedulePatientDiscovery({
+      requestId: uuidv7(),
+      patient,
+      source: MedicalDataSource.COMMONWELL,
+      facilityId,
+      rerunPdOnNewDemographics,
+      augmentDemographics,
+      isRerunFromNewDemographics,
+    });
+  } else if (statusCw === "processing" && scheduledPdRequestCw) {
```

Comment:
Couldnt we just remove this step and check if `statusCw` is not processing run the patient update

Reply:
Yes -- done. Had all cases explicitly while analyzing the code to make sure I covered everything.